### PR TITLE
React Course/Mocking Callbacks And Components/Add new article

### PR DESCRIPTION
**Pull request by Rllopez0410**
## Because
The additional resource can be helpful to some members struggling to understand act().

## This PR
- Included an article in the additional resources section.

## Issue

Closes [ #26435]( https://github.com/TheOdinProject/curriculum/issues/26435)

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: react/react_testing/mocking_callbacks_and_components.md [hunk 1]
```diff
@@ -166,4 +166,4 @@ The following questions are an opportunity to reflect on key topics in this less
 
 This section contains helpful links to related content. It isn't required, so consider it supplemental.
 
-- It looks like this lesson doesn't have any additional resources yet. Help us expand this section by contributing to our curriculum.
+- Dive deeper into [act()](https://codilime.com/blog/why-should-you-be-grateful-for-act-warnings-in-react-tests/) and understand its most common warning and when act() is not nessesary.
```
Comment (MaoShizhong) [inline react/react_testing/mocking_callbacks_and_components.md:169]:
```suggestion
- Dive deeper into [act()](https://codilime.com/blog/why-should-you-be-grateful-for-act-warnings-in-react-tests/) and understand its most common warning and when act() is not necessary.
```

**Review by MaoShizhong — CHANGES_REQUESTED** [review]

File: react/react_testing/mocking_callbacks_and_components.md (hunk 1 above)
Comment (MaoShizhong) [reply react/react_testing/mocking_callbacks_and_components.md:169]: @Rllopez0410 Did you mean to hit resolve? Just a little confused seeing the requested fix marked as resolve but no fix pushed. I've unmarked it now

Comment (Rllopez0410) [reply react/react_testing/mocking_callbacks_and_components.md:169]: Yeah, I thought I clicked on resolve. Sorry about that.

**Review by MaoShizhong — APPROVED** [review]
:rocket:

**Pull request merged**